cylc/flow/suite_files.py

```diff
@@ -602,6 +647,77 @@ def clean(reg):
     _remove_empty_reg_parents(reg, run_dir)
 
 
+def remote_clean(reg, platform_names):
+    """Run subprocesses to clean workflows on remote install targets
+    (skip localhost), given a set of platform names to look up.
+
+    Args:
+        reg (str): Workflow name.
+        platform_names (list): List of platform names to look up in the global
+            config, in order to determine the install targets to clean on.
+    """
+    try:
+        install_targets_map = (
+            get_install_target_to_platforms_map(platform_names))
+    except PlatformLookupError as exc:
+        raise PlatformLookupError(
+            "Cannot clean on remote platforms as the workflow database is "
+            f"out of date/inconsistent with the global config - {exc}")
+
+    queue = []
+    for target, platforms in install_targets_map.items():
+        if target == 'localhost':
+            continue
+        shuffle(platforms)
+        # Issue ssh command:
+        queue.append(
+            (_remote_clean_cmd(reg, platforms[0]), target, platforms)
+        )
+    failed_targets = []
+    for proc, target, platforms in queue:
+        ret_code = proc.wait()
+        out, err = (f.decode() for f in proc.communicate())
+        if out:
+            LOG.info(out)
+        if err:
+            LOG.warning(err)
+        if ret_code:
+            # Try again on the next platform for this install target:
+            this_platform = platforms.pop(0)
+            exc = TaskRemoteMgmtError(
+                TaskRemoteMgmtError.MSG_TIDY, this_platform['name'],
+                " ".join(proc.args), ret_code, out, err)
+            LOG.error(exc)
+            if platforms:
+                queue.append(
+                    (_remote_clean_cmd(reg, platforms[0]), target, platforms)
+                )
+            else:  # Exhausted list of platforms
+                failed_targets.append(target)
+    if failed_targets:
+        raise CylcError(
+            f"Could not clean on install targets: {', '.join(failed_targets)}")
+
+
+def _remote_clean_cmd(reg, platform):
+    """Remove a stopped workflow on a remote host.
+
+    Call "cylc clean --local-only" over ssh and return the subprocess.
+
+    Args:
+        reg (str): Workflow name.
+        platform (dict): Config for the platform on which to remove the
+            workflow.
+    """
+    LOG.info(
+        f'Cleaning on install target: {platform["install target"]} '
+        f'(platform: {platform["name"]})')
+    cmd = ['clean', '--local-only', reg]
+    cmd = construct_ssh_cmd(cmd, platform, timeout='10s')
```

Comment:
Sadly 10s is way too eager for our filesystems!

Not sure what a sensible value would be.

For interactive use we don't need a timeout if we capture ctrl+c then kill the SSH process.

Reply:
Note 1: In today's meeting we decided on 2 mins

Note 2: Reminder to self to look into using asyncio futures to paralellise waiting for each subprocess

Reply:
Should I include a cli option for the timeout in this PR?